### DISCUSSION TITLE
Questions regarding some specific parameters

### DISCUSSION THREAD

**Discussion by ggoetten** [openmopac/mopac#179] [Q&A]
Hello,

I have a couple of questions that I would love some answers.

First I would like to know if there is any information on precision of PM8, as it has been done to PM7 on the mopac website.

The second is that I am trying to run a polymer to calculate the vibration, but it says "Insufficient memory to run FORCE", I have 16 gb of Ram which is probably not much, and I am even using MOZYME recommended by the package. I guess there is nothing to be done correct? I even used FORCETS. It is very interesting that the energy minimization runs quite fine using PM7, RM1...

Lastly, I would like to know which kind of compounds THERMO should be used for, I tried to run some common drugs and none fits the criteria (paracetamol or caffeine). The system is only increasing as a straight line ? I don't know if it is my lack of background on physics but that doesn't fit the experimental data on Cp. No error is reported to the log file as well by mopac.

**MOPAC executable**
- [X] Using pre-built executable
- [ ] Compiled my own executable

**Operating system**
- [ ] Linux
- [ ] Mac
- [X] Windows
- [ ] Other (specify below)

**Additional context**
Add any other context about the problem here. (e.g. Linux distribution if that might be relevant)

**Comment by godotalgorithm** [comment]
I can answer these questions, and I've also made some changes to the development version of MOPAC in response to them (that will appear in the next MOPAC release).

1. There is no PM8 model yet, although it is in development (@Honza-R is leading the development as far as I understand). There is a placeholder PM8 model in MOPAC, but it is just a copy of PM6 at the moment. If the `PM8` keyword is used, I've added a note to the output file that the model is still in development.

2. There is an artificial size limiter on `FORCE` and `FORCETS` calculations, which I've removed from the development version. You should be able to get around the size limiter in older versions of MOPAC with `FORCETS` if you freeze the positions of many atoms in the system, which should then not count against the limit of 9000 vibrational degrees of freedom.

3. Beyond the discussion [in the manual](http://openmopac.net/manual/thermochemistry.html), I don't have a lot of expert advice to offer at the moment, as I haven't used the `THERMO` feature personally beyond some basic testing. I can at least say that it is certainly only applicable to the gas-phase form of whatever molecule you input. What do you mean by "none fits the criteria"? The two examples you mentioned (caffeine and paracetamol) have methyl groups with low-frequency rotational modes that may limit the accuracy of `THERMO`, but I'm not clear on the practical significance of this problem.

**Comment by Honza-R** [comment]
Please keep the PM8 model in the code - I'm using it for my parametrization experiments and it will turn into a real PM8 in the future. Keeping it in the code would make it easier to sync my development branch with the main version.

**Comment by zfekete2** [comment]
@ggoetten > Lastly, I would like to know which kind of compounds THERMO should be used for
 As @godotalgorithm has pointed out, the manual clearly states that there should be no internal rotors. This excludes many molecules of interest, such as those with methyl groups, alas.

 For a nice handle on this problem in general, you should looks at Grimme's work with their mRRHO method,
 [implemented in the CREST package](https://crest-lab.github.io/crest-docs/page/documentation/keywords.html#standalone-tools).
 With some (not-quite-trivial) tweaks it can be applied to the FORCE output from MOPAC. I've got some quick-and-dirty script for this, if you are interested.

**Comment by godotalgorithm** [comment]
The relevant reference for CREST's anharmonic corrections appears to be [[https://doi.org/10.1039/D1SC00621E](https://doi.org/10.1039/D1SC00621E)]. It uses a model that regularizes the free-energy divergences from low-frequency modes by approximating them as coming from rigid rotors, which is probably a good approximation for the rotations of small sp3-bonded side groups as in the original post.

A path to interfacing between MOPAC and CREST would certainly be welcome. I'm planning to add a directory in the MOPAC repo for pre/post-processing scripts, and I've received a few contributions this year over email. This script might fit in there.

**Comment by zfekete2** [comment]
@godotalgorithm > a good approximation for the rotations of small sp3-bonded side groups as in the original post.

 In fact mRRHO seems to handle much more, _e.g._ non-covalently bound complexes, as well. Indeed Grimme has a lot of earlier published work on it. The nice thing about CREST is that its THERMO module can be used as stand-alone processor for externally provided vibrational frequencies (so one needs not re-implement the algo in MOPAC, although that would be nice too).

 Regarding the script directory, having a dedicated place for collecting them is an excellent idea. I am too lazy/busy to learn something modern like Python for this, alas ;-), so my code is quaint in awk.
 But it works for me.

**Comment by Honza-R** [comment]
I do have an interface to MOPAC in my Cuby framework (
http://cuby4.molecular.cz/) where the rotor approximation can be applied
(using the keyword 'thermo_low_mode_fix'). At the moment, this may be the
most straightforward access to this combination of methods.

**Comment by zfekete2** [comment]
Cool! I've been tempted to try Cuby for quite some time, maybe this will push me over the edge to actually do it ;-)...

**Comment by godotalgorithm** [comment]
This Issue has been moved to a Discussion because it was resolved by clarifying existing features and did not constitute a bug report or new feature request.
